File: Programmers/LEVEL_1/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

bool compare(int a, int b)
{
    return a > b;
}

long long solution(long long n) {
    long long answer = 0;
    vector<int> v;
    int size = -1;

    while (n > 0)
    {
        int temp = n % 10;
        n /= 10;
        v.push_back(temp);
        size++;
    }

    sort(v.begin(), v.end(), compare);

    for (int i = 0; i < v.size(); i++)
    {
        answer += v[i] * pow(10, size);
        size--;
    }

    return answer;
}

File: Programmers/LEVEL_1/정수 내림차순으로 배치하기.js
function solution(n) {
  // var answer = 0;

  let temp = [];
  while (n > 0) {
    temp.push(n % 10);
    n = parseInt(n / 10, 10);
  }

  temp.sort((a, b) => b - a);

  var answer = parseInt(temp.join(""), 10);

  return answer;
}
